### PR TITLE
Update release pipeline to use Approvals and automate some manual tasks

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

1. Update the task based on environment approval to use Manual tasks 
2. Automate kicking off builds for release automation, vpack and snap
3. Automate adding change log to the GitHub release draft.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Review by TravisEz13 — APPROVED** [review]

File: tools/releaseBuild/azureDevOps/releasePipeline.yml [hunk 1]
```diff
@@ -139,24 +147,83 @@ stages:
...
+
+    - pwsh: |
+        Import-Module $(Agent.BuildDirectory)\Internal-PowerShellTeam-Tools\ReleaseTools\AzDO -Force
+        Set-AzDoAuthToken -Token $(RELEASEAUTOMATIONPAT)
+        Set-AzDoProjectInfo -ProjectOwner PowerShell-Rel -ProjectName Release-Automation
+        $packageBuildID = $(resources.pipeline.releasePipeline.runID)
+        $metadata = Get-Content -Raw -Path '$(Pipeline.Workspace)/releasePipeline/metadata/release.json' | ConvertFrom-Json
+        $buildInvocationInfo = Start-AzDOBuild -BuildDefinitionId 10 -BuildArguments @{ POWERSHELL_PACKAGE_BUILD_BUILDID = $packageBuildID } -Tag $metadata.ReleaseVersion, 'InProgress'
+        Write-Verbose -Verbose "Kicked off release automation: $($buildInvocationInfo.WebUrl)"
+        $status = $buildInvocationInfo | Wait-AzDOBuildStatus -Status Completed -TimeoutInMinutes 300
```
Comment (JamesWTruher) [inline tools/releaseBuild/azureDevOps/releasePipeline.yml:191]:
```suggestion
        $status = $buildInvocationInfo | Wait-AzDOBuildStatus -Status Completed -TimeoutInMinutes 300 -suppresProgress
```

File: tools/releaseBuild/azureDevOps/releasePipeline.yml [hunk 2]
```diff
@@ -139,24 +147,83 @@ stages:
...
+        if ($status.result -ne 'Succeeded') {
+          Write-Verbose "There are errors in release automation tests. Please triage failures."
```
Comment (JamesWTruher) [inline tools/releaseBuild/azureDevOps/releasePipeline.yml:193]: for the moment, you'll probably always get this message

File: tools/releaseBuild/azureDevOps/releasePipeline.yml [hunk 3]
```diff
@@ -139,24 +147,83 @@ stages:
...
+        artifact: metadata
+        path: '$(Pipeline.Workspace)/releasePipeline/metadata'
+
+    - pwsh: |
+        Import-Module $(Agent.BuildDirectory)\Internal-PowerShellTeam-Tools\ReleaseTools\AzDO -Force
+        Set-AzDoAuthToken -Token $(RELEASEAUTOMATIONPAT)
+        Set-AzDoProjectInfo -ProjectOwner PowerShell-Rel -ProjectName Release-Automation
+        $metadata = Get-Content -Raw -Path '$(Pipeline.Workspace)/releasePipeline/metadata/release.json' | ConvertFrom-Json
+        $azDOBuild = Get-AzDOBuild -buildDefinitionId 10 -MaximumResult 100  | Where-Object { $_.tags -contains $metadata.ReleaseVersion }
+        $azDoBuild | Remove-AzDOBuildTag -tag 'InProgress' -Pass | Add-AzDOBuildTag -Id $azDOBuild.Id -tag 'SignedOff'
```
Comment (JamesWTruher) [inline tools/releaseBuild/azureDevOps/releasePipeline.yml:225]: should there be a check for the job being complete?

File: tools/releaseBuild/azureDevOps/releasePipeline.yml [hunk 4]
```diff
@@ -280,19 +345,64 @@ stages:
...
+        artifact: metadata
+        path: '$(Pipeline.Workspace)/releasePipeline/metadata'
+    - pwsh: |
+        Import-Module $(Agent.BuildDirectory)\Internal-PowerShellTeam-Tools\ReleaseTools\AzDO -Force
+        Set-AzDoAuthToken -Token $(RELEASEAUTOMATIONPAT)
+        Set-AzDoProjectInfo -ProjectOwner PowerShell-Rel -ProjectName PowerShell
+        $metadata = Get-Content -Raw -Path '$(Pipeline.Workspace)/releasePipeline/metadata/release.json' | ConvertFrom-Json
+        $buildInvocationInfo = Start-AzDOBuild -BuildDefinitionId 49 -Tag $metadata.ReleaseVersion, 'InProgress'
+        Write-Verbose -Verbose "Kicked off snap build: $($buildInvocationInfo.WebUrl)"
+        $status = $buildInvocationInfo | Wait-AzDOBuildStatus -Status Completed -TimeoutInMinutes 60
```
Comment (JamesWTruher) [inline tools/releaseBuild/azureDevOps/releasePipeline.yml:369]: timeout seems short

File: tools/releaseBuild/azureDevOps/templates/release-CreateGitHubDraft.yml [hunk 5]
```diff
@@ -45,5 +45,37 @@ steps:
...
+
+    if (-not (Test-Path $filePath)) {
+      throw "$filePath not found"
+    }
+
+    $changelog = Get-Content -Raw -Path $filePath
+
+    $startPattern = "^## \[" + ([regex]::Escape($releaseVersion)) + "\]"
+    $endPattern = "^## \[{0}\.{1}\.{2}*" -f $semanticVersion.Major, $semanticVersion.Minor, $semanticVersion.Patch
+
```
Comment (JamesWTruher) [inline tools/releaseBuild/azureDevOps/templates/release-CreateGitHubDraft.yml:71]:
```suggestion
$outputLine = $false
```

**Review by JamesWTruher — APPROVED** [review]
mostly looks fine, but the `wait` cmdlet will emit progress

File: tools/releaseBuild/azureDevOps/releasePipeline.yml (hunk 4 above)
Comment (adityapatwardhan) [reply tools/releaseBuild/azureDevOps/releasePipeline.yml:369]: takes about 15-18 minutes

File: tools/releaseBuild/azureDevOps/releasePipeline.yml (hunk 2 above)
Comment (adityapatwardhan) [reply tools/releaseBuild/azureDevOps/releasePipeline.yml:193]: yup

File: tools/releaseBuild/azureDevOps/releasePipeline.yml (hunk 3 above)
Comment (adityapatwardhan) [reply tools/releaseBuild/azureDevOps/releasePipeline.yml:225]: waiting for completion in above task

**Review by PaulHigin — APPROVED** [review]

**Comment by TravisEz13** [comment]
Shouldn't be a full backport, but only of the manual approval changes

**Pull request merged**